### ISSUE TITLE
The position of PopupMenuBottom's menu is incorrect.

### ISSUE THREAD

**Issue by ipcjs** [flutter/flutter#69100]
the PR(#64678) caused this bug.

## Steps to Reproduce

1. run project [bug.zip](https://github.com/flutter/flutter/files/5444871/bug.zip)
2. click PopupMenuButton in AppBar

**Expected results:** 
the top of the menu and PopupMenuButton are aligned.

**Actual results:** 
On the windows platform, the display is correct.
On the android platform, the menu is offset downward.

screenshot on windows:
![Image](https://user-images.githubusercontent.com/4396864/97295557-6ab91500-188a-11eb-8c41-bb7d24755fe6.png)

screenshot on android:
![Image](https://user-images.githubusercontent.com/4396864/97295682-963bff80-188a-11eb-9156-d0066f6be9ef.png)

<details>
  <summary>flutter doctor -v</summary>

```
[✓] Flutter (Channel dev, 1.24.0-3.0.pre, on Microsoft Windows [Version 10.0.19041.572], locale zh-CN)
    • Flutter version 1.24.0-3.0.pre at C:\SDK\flutter
    • Framework revision 2783f8e2e1 (5 days ago), 2020-10-22 09:36:06 -0700
    • Engine revision defa8be2b1
    • Dart version 2.11.0 (build 2.11.0-242.0.dev)
    • Pub download mirror https://pub.flutter-io.cn
    • Flutter download mirror https://storage.flutter-io.cn

[✓] Android toolchain - develop for Android devices (Android SDK version 29.0.3)
    • Android SDK at C:\SDK\android-sdk
    • Platform android-30, build-tools 29.0.3
    • ANDROID_HOME = C:\SDK\android-sdk
    • Java binary at: C:\SDK\android-studio-beta\jre\bin\java
    • Java version OpenJDK Runtime Environment (build 1.8.0_242-release-1644-b01)
    • All Android licenses accepted.

[✓] Chrome - develop for the web
    • Chrome at C:\Program Files (x86)\Google\Chrome\Application\chrome.exe

[✓] Visual Studio - develop for Windows (Visual Studio Community 2019 16.4.2)
    • Visual Studio at C:\Program Files (x86)\Microsoft Visual Studio\2019\Community
    • Visual Studio Community 2019 version 16.4.29613.14
    • Windows 10 SDK version 10.0.18362.0

[✓] Android Studio (version 4.1.0)
    • Android Studio at C:\SDK\android-studio-beta
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 1.8.0_242-release-1644-b01)

[✓] IntelliJ IDEA Community Edition (version 2020.2)
    • IntelliJ at C:\Users\ipcjs\AppData\Local\JetBrains\Toolbox\apps\IDEA-C\ch-0\202.4357.23
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart

[✓] VS Code (version 1.50.1)
    • VS Code at C:\Users\ipcjs\AppData\Local\Programs\Microsoft VS Code
    • Flutter extension version 3.15.1

[✓] Proxy Configuration
    • HTTP_PROXY is set
    • NO_PROXY is localhost,127.0.0.1,dl.google.com,maven.aliyun.com,cdn.npm.taobao.org,registry.npm.taobao.org,storage.flutter-io.cn,pub.flutter-io.cn,services.gradle.org,apple.com,downloads.gradle-dn.com
    • NO_PROXY contains 127.0.0.1
    • NO_PROXY contains localhost

[✓] Connected device (5 available)
    • SM G960U (mobile) • 433943354c313098 • android-arm64  • Android 10 (API 29)
    • Windows (desktop) • windows          • windows-x64    • Microsoft Windows [Version 10.0.19041.572]
    • Web Server (web)  • web-server       • web-javascript • Flutter Tools
    • Chrome (web)      • chrome           • web-javascript • Google Chrome 86.0.4240.111
    • Edge (web)        • edge             • web-javascript • Microsoft Edge 86.0.622.51

• No issues found!
```

</details>

**Comment by darshankawar** [comment]
@ipcjs 
I am unable to replicate this using latest stable version on Pixel 3XL emulator (api 30) and on Android Samsung 8.1.0 device.

![69100](https://user-images.githubusercontent.com/67046386/97305202-d0090800-1882-11eb-8449-a68aa6c497cf.png)

<img width="344" alt="Screenshot 2020-10-27 at 6 29 13 PM" src="https://user-images.githubusercontent.com/67046386/97305211-d26b6200-1882-11eb-8dfc-e1caaf54552a.png">


<details>
<summary>flutter doctor -v</summary>

```
[✓] Flutter (Channel stable, 1.22.2, on Mac OS X 10.15.4 19E2269, locale en-IN)
    • Flutter version 1.22.2 at /Users/dhs/documents/Fluttersdk/flutter
    • Framework revision 84f3d28555 (12 days ago), 2020-10-15 16:26:19 -0700
    • Engine revision b8752bbfff
    • Dart version 2.10.2

 
[✓] Android toolchain - develop for Android devices (Android SDK version 30.0.0)
    • Android SDK at /Users/dhs/Library/Android/sdk
    • Platform android-30, build-tools 30.0.0
    • Java binary at: /Applications/Android
      Studio.app/Contents/jre/jdk/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build
      1.8.0_242-release-1644-b3-6222593)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 12.0.1)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Xcode 12.0.1, Build version 12A7300
    • CocoaPods version 1.9.3

[✓] Android Studio (version 4.0)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin version 46.0.2
    • Dart plugin version 193.7361
    • Java version OpenJDK Runtime Environment (build
      1.8.0_242-release-1644-b3-6222593)

[✓] VS Code (version 1.50.1)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.15.1

[✓] Connected device (2 available)
    • SM A260G (mobile)           • 5200763ebcfa861f • android-arm • Android
      8.1.0 (API 27)
    • sdk gphone x86 arm (mobile) • emulator-5554    • android-x86 • Android 11
      (API 30) (emulator)

• No issues found!


```

</details>

Can you switch to latest master or stable and check if you experience same behavior ?
Thanks.

**Comment by ipcjs** [comment]
the PR(#64678) caused this bug. It has been merged into the master and dev branches.
I am using the latest dev branch.

**Comment by darshankawar** [comment]
Issue reproducible on latest master but not on stable (per my earlier [comment](https://github.com/flutter/flutter/issues/69100#issuecomment-717228480) and screenshots). 

![69100_master](https://user-images.githubusercontent.com/67046386/97311944-f16df200-188a-11eb-8b06-79e8786687f8.png)

<details>
<summary>flutter doctor -v</summary>

```
[✓] Flutter (Channel master, 1.24.0-4.0.pre.82, on Mac OS X 10.15.4 19E2269
    darwin-x64, locale en-IN)
    • Flutter version 1.24.0-4.0.pre.82 at
      /Users/dhs/documents/Fluttersdk/flutter
    • Framework revision e5814756a2 (5 hours ago), 2020-10-27 01:47:03 -0700
    • Engine revision 1857470267
    • Dart version 2.11.0 (build 2.11.0-260.0.dev)

[✓] Android toolchain - develop for Android devices (Android SDK version 30.0.0)
    • Android SDK at /Users/dhs/Library/Android/sdk
    • Platform android-30, build-tools 30.0.0
    • Java binary at: /Applications/Android
      Studio.app/Contents/jre/jdk/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build
      1.8.0_242-release-1644-b3-6222593)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 12.0.1)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Xcode 12.0.1, Build version 12A7300
    • CocoaPods version 1.9.3

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 4.0)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin version 46.0.2
    • Dart plugin version 193.7361
    • Java version OpenJDK Runtime Environment (build
      1.8.0_242-release-1644-b3-6222593)

[✓] VS Code (version 1.50.1)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.15.1

[✓] Connected device (4 available)
    • SM A260G (mobile) • 5200763ebcfa861f • android-arm    • Android 8.1.0 (API
      27)
    • macOS (desktop)   • macos            • darwin-x64     • Mac OS X 10.15.4
      19E2269 darwin-x64
    • Web Server (web)  • web-server       • web-javascript • Flutter Tools
    • Chrome (web)      • chrome           • web-javascript • Google Chrome
      86.0.4240.111

• No issues found!


```

</details>

Could be due to [this](https://github.com/flutter/flutter/pull/64678) PR

**Comment by xu-baolin** [comment]
This is indeed break material design: https://material.io/components/menus#dropdown-menu
![image](https://user-images.githubusercontent.com/61075224/97398615-7bb76400-1926-11eb-8c2a-89f2878fbc96.png)

I am working on this issue.

**Comment by xu-baolin** [comment]
Almost finish coding, but depend on this PR #69383

**Comment by MrBirb** [comment]
Looks like PR was reverted. This should be reopened.

**Comment by JagandeepBrar** [comment]
This should be reopened again, as the reland has also been reverted.

**Comment by xu-baolin** [comment]
Fixed via #78395
